netlogo-core/src/test/compile/api/PrimitiveMungersTests.scala

```diff
@@ -0,0 +1,172 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.compile.api
+
+import org.nlogo.core.{ Command => CoreCommand, Reporter => CoreReporter,
+  SourceLocation, Syntax }
+import org.nlogo.core.prim.{ _const => _coreconst }
+import org.nlogo.nvm.{ Command, Context, Instruction, Reporter }
+import org.scalatest.{ FunSuite, Inside }
+
+object PrimitiveMungersTests {
+  class DummyCoreCommand(rights: Int*) extends CoreCommand {
+    def syntax = Syntax.commandSyntax(right = rights.toList)
+  }
+  class DummyCoreReporter extends CoreReporter {
+    def syntax = Syntax.reporterSyntax(ret = Syntax.NumberType)
+  }
+  class DummyCommand extends Command {
+    def perform(c: Context): Unit = {}
+  }
+  class DummyReporter extends Reporter {
+    def report(c: Context): AnyRef = Double.box(0.0)
+  }
+}
+
+import PrimitiveMungersTests._
+
+class PrimitiveMungersTests extends FunSuite with Inside {
+
+  def newMatch(a: AstNode): Match = {
+    new Match(a)
+  }
+
+  val constNum = {
+    new ReporterApp(new _coreconst(Double.box(0.0)), new DummyReporter(), Seq(), new SourceLocation(0, 0, ""))
+  }
+
+  def statement(cc: CoreCommand, c: Command, args: Expression*): Statement = {
+    new Statement(cc, c, args, new SourceLocation(0, 0, ""))
+  }
+
+  def repApp(cr: CoreReporter, r: Reporter): ReporterApp = {
+    new ReporterApp(cr, r, Seq(), new SourceLocation(0, 0, ""))
+  }
+
+  test("match.command returns the statement command") {
+    val command = new DummyCommand()
```

Comment:
I mentioned yesterday that there is a way to pull out `val command = new DummyCommand()` from all of these. Use what scalatest refers to as a "fixture-context object" [see here for details](http://doc.scalatest.org/3.0.1/index.html#org.scalatest.FunSuite@fixtureContextObjects). We can discuss tomorrow.